### DISCUSSION TITLE
Why can't I extend the biome class?

### DISCUSSION THREAD

**Discussion by Flamgop** [Minestom/Minestom#680] [Help] [answered]
I would like to be able to extend the biome class, in case I would like to add methods or properties such as biome specific populators or biome specific noise types. I can get around it easily its just kind of annoying.

**Comment by TheMode** [comment, marked as answer]
Because there is no reason to and would prevent us from making them part of the registry API.

Biomes are stored as numerical id, and there is no guarantee that your misc data can be properly/efficiently serialised.

**Reply by Flamgop** [reply]
Oh ok that makes sense lol.
